csharp/src/Ice/Reference.cs

```diff
@@ -1014,26 +1035,110 @@ void TraceRetry(string message, int attempt, RetryPolicy policy, Exception? exce
                         $"received proxy for protocol {protocol.GetName()} with invocation mode set");
                 }
 
-                // The min size for an Endpoint with the 2.0 encoding is: transport (short = 2 bytes) + host name
-                // (min 2 bytes as it cannot be empty) + port number (ushort, 2 bytes) + options (1 byte for empty
-                // sequence), for a total of 7 bytes.
-                IReadOnlyList<Endpoint> endpoints = proxyKind == ProxyKind.Direct ?
-                    istr.ReadArray(minElementSize: 7, istr => istr.ReadEndpoint(protocol)) :
-                    ImmutableArray<Endpoint>.Empty;
+                if (proxyKind == ProxyKind20.IndirectRelative)
+                {
+                    if (istr.Connection != null)
+                    {
+                        Communicator communicator = istr.Connection!.Communicator;
```

Comment:
you can declare this bellow when we first use it, you don't need ! to access the connection as we just checked for null when entering the if

Reply:
Fixed.